api/applications/views.py

```diff
@@ -97,3 +104,31 @@ def perform_update(self, serializer):
         serializer.validated_data['owner'] = self.request.user
         # TODO: Write code to transfer ownership
         serializer.save(owner=self.request.user)
+
+
+class ApplicationReset(generics.UpdateAPIView, ApplicationMixin):
```

Comment:
Discussed with @brianjgeiger : Since the route is a custom action (`/<id>/reset/`), we should use `POST` for this route. AFAIK, `UpdatAPIView` implements `PUT` and `PATCH`. This should be changed to a generic view class that implements `POST`.